packages/@aws-cdk/aws-codebuild/test/linux-arm-build-image.test.ts

```diff
@@ -0,0 +1,361 @@
+          computeType: codebuild.ComputeType.LARGE,
+          buildImage: codebuild.LinuxArmBuildImage.AMAZON_LINUX_2_ARM_2,
+        },
+      });
+
+      Template.fromStack(stack).hasResourceProperties('AWS::CodeBuild::Project', {
+        Environment: {
+          Type: 'ARM_CONTAINER',
+          ComputeType: 'BUILD_GENERAL1_LARGE',
+        },
+      });
+    });
+
+    test('cannot be used in conjunction with ComputeType X2_LARGE', () => {
+      const stack = new cdk.Stack();
+
+      expect(() => {
+        new codebuild.PipelineProject(stack, 'Project', {
+          environment: {
+            buildImage: codebuild.LinuxArmBuildImage.AMAZON_LINUX_2_ARM_2,
+            computeType: codebuild.ComputeType.X2_LARGE,
+          },
+        });
+      }).toThrow(/ARM images only support ComputeTypes 'BUILD_GENERAL1_SMALL' and 'BUILD_GENERAL1_LARGE' - 'BUILD_GENERAL1_2XLARGE' was given/);
+    });
+  });
+
+  describe('ECR Repository', () => {
+    test('allows creating a build image from a new ECR repository', () => {
+      const stack = new cdk.Stack();
+
+      const repository = new ecr.Repository(stack, 'my-repo');
+
+      new codebuild.Project(stack, 'Project', {
+        buildSpec: codebuild.BuildSpec.fromObject({
+          version: '0.2',
+          phases: {
+            build: { commands: ['ls'] },
+          },
+        }),
+        environment: {
+          buildImage: codebuild.LinuxArmBuildImage.fromEcrRepository(repository, 'v1'),
+        },
+      });
+
+      Template.fromStack(stack).hasResourceProperties('AWS::CodeBuild::Project', {
+        Environment: {
+          ComputeType: 'BUILD_GENERAL1_LARGE',
+          Image: {
+            'Fn::Join': [
+              '',
+              [
+                {
+                  'Fn::Select': [
+                    4,
+                    {
+                      'Fn::Split': [
+                        ':',
+                        {
+                          'Fn::GetAtt': [
+                            'myrepo5DFA62E5',
+                            'Arn',
+                          ],
+                        },
+                      ],
+                    },
+                  ],
+                },
+                '.dkr.ecr.',
+                {
+                  'Fn::Select': [
+                    3,
+                    {
+                      'Fn::Split': [
+                        ':',
+                        {
+                          'Fn::GetAtt': [
+                            'myrepo5DFA62E5',
+                            'Arn',
+                          ],
+                        },
+                      ],
+                    },
+                  ],
+                },
+                '.',
+                {
+                  Ref: 'AWS::URLSuffix',
+                },
+                '/',
+                {
+                  Ref: 'myrepo5DFA62E5',
+                },
+                ':v1',
+              ],
+            ],
```

Comment:
Let's try to preserve vertical space where we can:

```suggestion
            'Fn::Join': ['', [
              {
                'Fn::Select': [4, {
                  'Fn::Split': [':', {
                    'Fn::GetAtt': ['myrepo5DFA62E5', 'Arn'],
                  }],
                }],
              },
              '.dkr.ecr.',
              {
                'Fn::Select': [3, {
                  'Fn::Split': [':', {
                    'Fn::GetAtt': [ 'myrepo5DFA62E5', 'Arn'],
                  }],
                }],
              },
              '.',
              { Ref: 'AWS::URLSuffix' },
              '/',
              { Ref: 'myrepo5DFA62E5' },
              ':v1',
            ]],
```